### PR TITLE
#3xhth - Added documentation on how to continue development

### PR DESCRIPTION
**Pull request by wilbo**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Scala
 target
 *.class
+src/main/resources
```
Comment (jjkramok) [inline .gitignore]: Isn't the resources directory meant to be tracked?

File: documentation/TechnicalDesignDocument.tex [hunk 2]
```diff
@@ -96,6 +96,26 @@
...
     How we applied it, the software used and the targets and platforms supported are among a few.
 
+    %------------------------------------------------------------------------------------------------------------------------------------------------------- Continuing Development
+    \section{Continuing Development}
+    This section describes the plan on how to continue with development in the best way.
+    The team worked on quite some user stories from april 9th, to june 15th.
+    The following functionalities are most critical at the moment to start working on:
+    \myparagraph{Instantiating maramafications from a .mar file.}
+    A beginning is made at implementing the .mar file.
+    The .mar file is basically a json file with some agreements on structure.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: Isn't a .mar file a .obj file with a json header?

File: documentation/TechnicalDesignDocument.tex [hunk 3]
```diff
@@ -96,6 +96,26 @@
...
+    The goal of the .mar file is, whenever the file is loaded in, a maramafication is instantiated.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: Not sure if this is clear. But as far as I know it just means it is added to the list of all possible Ms (think the sidebar in the view project).

File: documentation/TechnicalDesignDocument.tex [hunk 4]
```diff
@@ -96,6 +96,26 @@
...
+    This means that every value from the .mar file is evaluated and constructed into a usable maramafication model in the View.
+    \myparagraph{Communication from Editor to View}
+    The Editor must be able to send its state to the View so the View can visually represent that state.
+    This also requires some work inside the View to make this happen.
+    The reason behind this is to keep the Editor end View separated.
+    This wat the Editor and View have their own goal.
+    \myparagraph{Saving the state of the View inside the Editor}
+    The Editor needs to be able to know the state that the View is in.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: Isn't it the other way around? The editor should contain the state and the view should only display stuff and redirect input to the editor

File: documentation/TechnicalDesignDocument.tex [hunk 5]
```diff
@@ -96,6 +96,26 @@
...
+    This is because only the Editor is able to know whether a current marama is valid or invalid.
```
Comment (jjkramok) [inline documentation/TechnicalDesignDocument.tex]: The view should query the Editor if a move / command is legal

**Comment by jjkramok** [comment]
There is no update to the review. Currently the only thing in there tells a person that it needs some version of Scala and some version of sbt and how to build an artifact. This could be more precise and I think this is not everything.

File: documentation/TechnicalDesignDocument.tex (hunk 2 above)
Comment (wilbo) [reply documentation/TechnicalDesignDocument.tex]: Nope, from my understanding the .obj files are linked or embedded from inside the .mar file. Is this corrent @ThijsvanDam?

File: documentation/TechnicalDesignDocument.tex (hunk 3 above)
Comment (wilbo) [reply documentation/TechnicalDesignDocument.tex]: Added a clarification.

File: README.md [hunk 6]
```diff
@@ -1,8 +1,17 @@
...
+
+## Development
+
+This project is based on Scala and sbt. To start developing you can import the project by opening the build.sbt file inside the project root with an IDE like IntelliJ or Eclipse.
+
+## Documentation
+
+It is recommended before you start development to read the technical design document that is provided in the source. 
+
+[TODO: describe how to build technical design document]
```
Comment (LuneCoding) [inline README.md]: TODO

File: documentation/TechnicalDesignDocument.tex (hunk 3 above)
Comment (LuneCoding) [reply documentation/TechnicalDesignDocument.tex]: The goal of the .mar file is to instantiate a maramafication when the file is loaded in.

Maybe?

File: README.md (hunk 6 above)
Comment (LuneCoding) [reply README.md]: Also read the devlog?

File: documentation/TechnicalDesignDocument.tex [hunk 7]
```diff
@@ -96,6 +96,27 @@
...
+    A beginning is made at implementing the .mar file.
+    The .mar file is basically a json file with some agreements on structure.
+    The goal of the .mar file is, whenever the file is loaded in, a maramafication is instantiated.
+    This means that every value from the .mar file is evaluated and constructed into a usable maramafication model in the View.
+    This means that it will be placed in the maramafication list of the View.
+    \myparagraph{Communication from Editor to View}
+    The Editor must be able to send its state to the View so the View can visually represent that state.
+    This also requires some work inside the View to make this happen.
+    The reason behind this is to keep the Editor and View separated.
+    This way the Editor and View have their own goal.
```
Comment (LuneCoding) [inline documentation/TechnicalDesignDocument.tex]: Why do they have a separate goal?

File: README.md (hunk 6 above)
Comment (wilbo) [reply README.md]: True but that is not really a concern for this repo. De documentation is inside this repo so it should be read before starting development.

File: documentation/TechnicalDesignDocument.tex (hunk 7 above)
Comment (wilbo) [reply documentation/TechnicalDesignDocument.tex]: removed this useless sentence

File: README.md (hunk 6 above)
Comment (LuneCoding) [reply README.md]: I'd recommend fully removing the TODO, let them figure it out for themselves.

Comment (wilbo) [reply README.md]: Ok, removed.

**Pull request merged**